lib/netext/httpext/tracer_test.go

```diff
@@ -216,20 +216,21 @@ func TestCancelledRequest(t *testing.T) {
 	cancelTest := func(t *testing.T) {
 		t.Parallel()
 		tracer := &Tracer{}
-		req, err := http.NewRequest("GET", srv.URL+"/delay/1", nil)
+		req, err := http.NewRequestWithContext(context.Background(), "GET", srv.URL+"/delay/1", nil)
 		require.NoError(t, err)
 
 		ctx, cancel := context.WithCancel(httptrace.WithClientTrace(req.Context(), tracer.Trace()))
 		req = req.WithContext(ctx)
 		go func() {
-			time.Sleep(time.Duration(rand.Int31n(50)) * time.Millisecond)
+			time.Sleep(time.Duration(rand.Int31n(50)) * time.Millisecond) //nolint:gosec
 			cancel()
 		}()
 
-		resp, err := srv.Client().Transport.RoundTrip(req)
+		resp, err := srv.Client().Transport.RoundTrip(req) //nolint:bodyclose
 		_ = tracer.Done()
-		assert.Nil(t, resp)
-		assert.Error(t, err)
+		if resp == nil && err == nil {
+			t.Errorf("Expected either a RoundTrip response or error or trail errors but got %#v and %#v", resp, err)
```

Comment:
Ok, so this test is never going to fail? :confused:

Reply:
well, that is what I have been trying to explain ... that this is test is strange ... before I "fixed" it it was doing the exact same thing ... 
As far as I could see you tried to tease out a race condition with this ... and just it so happens that it breaks(IMO) ... strangely and so you wrote it in a way ... that it checks at least that it either errors out or returns an actual response ... which IMO isn't a big check but it's ... something :shrug:

Reply:
I'm also puzzled by this test... Could the `rand.Int31n(50)` be causing issues if it returns 0? Maybe we need a minimum sleep period there.

Also the "or trail errors" part should be removed from the error message since it's no longer checking for that.

Reply:
> Could the rand.Int31n(50) be causing issues if it returns 0

I reverted it back to the rand ... previously I was sleeping 100ms every time - no difference